draft-normington-jsonpath-latest.xml

```diff
@@ -104,7 +107,40 @@
       </section>
     </section>
    <section numbered="true" toc="default">
-      <name>JSONPath Syntax and Semantics</name>         
+      <name>JSONPath Syntax and Semantics</name>
+         <section numbered="true" toc="default">
+            <name>Node Matching Semantics</name>
+            <t>A JSONPath is a string which matches zero or more nodes of any JSON document.
+            </t>
+            <t>A JSONPath is abstractly a sequence of matchers. Each matcher takes a list of nodes and produces a possibly different list of nodes.
```

Comment:
> A JSONPath is abstractly a sequence of matchers.

Things getting not that simple if we support subpaths and/or nested filters, and we have [a consensus](https://cburgmer.github.io/json-path-comparison/results/filter_expression_with_subpaths.html) here.

Reply:
@remorhaz I'm not sure I follow. If we take that example `$[?(@.address.city=='Berlin')]`, that will break down into two matchers:
* `$` which matches the root document
*  `[?(@.address.city=='Berlin')]` which filters the array to pick out the "Berlin" element.
Some of the matchers will be defined in terms of others, but even so each JSONPath will correspond to a sequence of "top-level" matchers.

Reply:
Well, if we don't speak about "inner" matchers here, then I agree with the definition.